src/Arcus.Security.Providers.KeyPerFile/SecretStoreBuilderExtensions.cs

```diff
@@ -0,0 +1,19 @@
+﻿using Microsoft.Extensions.Configuration.KeyPerFile;
+using Microsoft.Extensions.FileProviders;
+using Microsoft.Extensions.Hosting;
+
+namespace Arcus.Security.Providers.KeyPerFile
+{
+    public static class SecretStoreBuilderExtensions
+    {
+        public static SecretStoreBuilder AddKeyPerFile(this SecretStoreBuilder builder, string directoryPath, bool optional = true)
+        {
+            KeyPerFileConfigurationSource configuration = new KeyPerFileConfigurationSource();
```

Comment:
```suggestion
        {
            Guard.NotNull(builder, nameof(builder), "Requires a secret store builder to add the Docker secrets to");
            Guard.NotNullOrWhitespace(directoryPath, nameof(directoryPath), "Requires a non-blank directory path to locate the Docker secrets");
            
            KeyPerFileConfigurationSource configuration = new KeyPerFileConfigurationSource();
```

Let's guard here.

Reply:
Done.   I did it via VS.NET instead of committing the suggestion, since I've renamed some files and directories.